File: evidence/generated-systems/golang/aerospace-on-golang/aerospace-on-golang/internal/model/AircraftPackage.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// AircraftPackage Declaration
//==============================================================
type AircraftPackage struct {
    gorm.Model
     Name                                    string
     Options           []AircraftOption `gorm:"foreignKey:OptionsFromAircraftPackageId"`
     Variants           []AircraftVariant `gorm:"foreignKey:VariantsFromAircraftPackageId"`
    PackageType                      PackageType

// parent associations as their child

}

File: evidence/generated-systems/django/aerospace-on-django/aerospaceOnDjango/models/AircraftPackage.py
from django.db import models
from aerospaceOnDjango.models.PackageType import PackageType

#======================================================================
# 
# Encapsulates data for model AircraftPackage
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class AircraftPackage Declaration
#======================================================================
class AircraftPackage (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	options = models.ManyToManyField('AircraftOption',  blank=True, related_name='+')
	variants = models.ManyToManyField('AircraftVariant',  blank=True, related_name='+')
	packageType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in PackageType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.packageType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "AircraftPackage";
    
	def objectType(self):
		return "AircraftPackage";
